### DISCUSSION TITLE
Upgrade error - failed to register layer

### DISCUSSION THREAD

**Discussion by spideyNTX** [jasonacox/Powerwall-Dashboard#413] [Q&A] [answered]
Anyone ever received this error during an upgrade?
`failed to register layer: Error processing tar file(exit status 1): archive/tar: invalid tar header`

I upgraded from 2.10.0 to 3.0.7. Git changes seem to take ok. Timezone set back correctly. 
Then got to this:
```
Updating Powerwall-Dashboard stack...
Running Docker Compose...
[+] Running 8/9
   ⠙ pypowerwall 8 layers [⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulling     4.2s
   ✔ 661ff4d9561e Pull complete                                                 0.3s
   ✔ 44cda88cd45d Pull complete                                                0.2s
   ✔ 48c91875651e Download complete                                      0.4s
   ✔ 4a3bf9ff3965 Download complete                                        0.4s
   ✔ 8c653295a5ef Download complete                                      2.2s
   ✔ 90637a1715f1 Download complete                                      0.6s
   ✔ 0d06604adf74 Download complete                                     2.5s
   ✔ bf969b869dc7 Download complete                                     2.2s
failed to register layer: Error processing tar file(exit status 1): archive/tar: invalid tar header
```

**Comment by jasonacox** [comment, marked as answer]
Hi @spideyNTX - What OS are you using? I can't replicate that on Linux, RPI, Windows (WSL) or MacOS. I know that error can show up on Windows on Powershell based on a quick search on Google but I can't find anything else. It could also just be a network glitch with DockerHub during the download. Try again and  possibly remove all the existing pypowerwall images from docker (e.g. `docker images` / `docker rmi`).

**Reply by spideyNTX** [reply]
Debian 12.  Must have been a temporary issue retrieving updated docker images as you suggested.  I ran upgrade.sh again, git detected all files where already in sync, and this time all images were downloaded/pulled successfully, and the process completed.
